crates/subspace-farmer/src/bin/subspace-farmer/commands/farm/dsn.rs

```diff
@@ -314,23 +357,21 @@ pub(crate) fn start_announcements_processor(
     node: Node,
     piece_cache: Arc<tokio::sync::Mutex<FarmerPieceCache>>,
     weak_readers_and_pieces: Weak<Mutex<Option<ReadersAndPieces>>>,
+    record_processor_helper: RecordProcessorAnnouncementHelper,
 ) -> io::Result<HandlerId> {
     let (provider_records_sender, mut provider_records_receiver) =
         mpsc::channel(MAX_CONCURRENT_ANNOUNCEMENTS_QUEUE.get());
 
-    let handler_id = node.on_announcement(Arc::new({
+    let handler_id = record_processor_helper.on_announcement(Arc::new({
```

Comment:
What if we remove `record_processor_helper` and instead have `provider_records_receiver` as an argument of this function and sender can be called from `PieceAnnouncementRequestHandler::create`? I think it might be easier to follow that way.

Reply:
I'm not sure I understood 100% your proposal. The latest commit is the best effort to address your suggestion.

Reply:
My point was that we don't need these abstractions. We can remove both `RecordProcessorAnnouncementHelper` and related type aliases.